pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -298,10 +300,16 @@ func toEnvExtensionDataModel(e ExtensionClassification) datamodel.Extension {
 func toEnvironmentRecipeProperties(e RecipePropertiesClassification) (datamodel.EnvironmentRecipeProperties, error) {
 	switch c := e.(type) {
 	case *TerraformRecipeProperties:
+		// Radius only supports Terraform registry and HTTP URLs as module sources.
 		if c.TemplatePath != nil {
-			// Check for local paths
-			if strings.HasPrefix(to.String(c.TemplatePath), "/") || strings.HasPrefix(to.String(c.TemplatePath), "./") || strings.HasPrefix(to.String(c.TemplatePath), "../") {
-				return datamodel.EnvironmentRecipeProperties{}, v1.NewClientErrInvalidRequest(fmt.Sprintf(invalidLocalModulePathFmt, to.String(c.TemplatePath)))
+			// We first validate if the template path is a valid Terraform registry module source.
+			_, err := tfaddr.ParseModuleSource(to.String(c.TemplatePath))
+			if err != nil {
+				// If the template path is not a Terraform registry module source, we validate if it is an HTTP URL.
+				_, err = url.ParseRequestURI(to.String(c.TemplatePath))
+				if err != nil {
+					return datamodel.EnvironmentRecipeProperties{}, v1.NewClientErrInvalidRequest(fmt.Sprintf(invalidModulePathFmt, to.String(c.TemplatePath)))
+				}
```

Comment:
Should this be a separate function? Is this going to be used elsewhere in the code?

Reply:
No, this is just going to be used here when we try to register a recipe to an environment. We don't want to deploy the resource/recipe at all if we know the module source isn't supported and this is the first entry point for registering the recipe